tests/test_writer.py

```diff
@@ -557,6 +558,59 @@ def test_invalid_transformations(self, coordinateTransformations):
         with pytest.raises(ValueError):
             write_multiscales_metadata(self.root, datasets, axes=axes)
 
+    @pytest.mark.parametrize(
+        "metadata",
+        [
+            {
+                "channels": [
+                    {
+                        "color": "FF0000",
+                        "window": {"start": 0, "end": 255, "min": 0, "max": 255},
+                    }
+                ]
+            },
+            {"channels": [{"color": "FF0000"}]},
+            {
+                "channels": [  # test wrong metadata
+                    {
+                        "color": "FF0000",
+                        "window": {"start": 0, "end": 255, "min": 0},
+                    }
+                ]
+            },
+            None,
+        ],
+    )
+    def test_omero_metadata(self, metadata: Optional[Dict[str, Any]]):
+        datasets = []
+        for level, transf in enumerate(TRANSFORMATIONS):
+            datasets.append({"path": str(level), "coordinateTransformations": transf})
+        if metadata is None:
+            with pytest.raises(
+                KeyError, match="If `'omero'` is present, value cannot be `None`."
+            ):
+                write_multiscales_metadata(
+                    self.root, datasets, axes="tczyx", metadata={"omero": metadata}
+                )
+        elif "window" in metadata["channels"][0]:
+            window_metadata = metadata["channels"][0].get("window")
+            if len(window_metadata) < 4:
+                with pytest.raises(KeyError, match=".*`'window'`.*"):
+                    write_multiscales_metadata(
+                        self.root, datasets, axes="tczyx", metadata={"omero": metadata}
+                    )
+        elif "color" in metadata["channels"][0]:
```

Comment:
Just noticed that with `elif` here, you'll only test the `color` validation if there is no `window` metadata.
Just use `if "color"..` instead.
But also I don't see that `if len(color_metadata) < 6:` below is ever true, since all the `color` in `parametrize` above are length `6`.